spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/outbound/HazelcastOutboundChannelAdapterTests.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.outbound;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import javax.annotation.Resource;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.DefaultMessageBuilderFactory;
+import org.springframework.integration.support.MessageBuilderFactory;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Hazelcast Outbound Channel Adapter Test Class
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+@DirtiesContext
+public class HazelcastOutboundChannelAdapterTests {
+
+	private static final int DATA_COUNT = 100;
+
+	private volatile MessageBuilderFactory messageBuilderFactory;
+
+	@Autowired
+	private MessageChannel mapChannel;
+
+	@Autowired
+	private MessageChannel listChannel;
+
+	@Autowired
+	private MessageChannel setChannel;
+
+	@Autowired
+	private MessageChannel queueChannel;
+
+	@Resource
+	private Map<?, ?> distributedMap;
+
+	@Resource
+	private List<?> distributedList;
+
+	@Resource
+	private Set<?> distributedSet;
+
+	@Resource
+	private Queue<?> distributedQueue;
+
+	@Before
+	public void setUp() {
+		distributedMap.clear();
+		distributedList.clear();
+		distributedSet.clear();
+		distributedQueue.clear();
+
+		messageBuilderFactory = new DefaultMessageBuilderFactory();
+	}
+
+	@Test
+	public void testWriteDistributedMap() {
+		Map<Integer, String> map = createMapByEntryCount();
+		mapChannel.send(this.messageBuilderFactory.withPayload(map).build());
```

Comment:
No reason to use `messageBuilderFactory` in tests and even `MessageBuilder`, if you get deal only with `payload`. 
```
new GenericMessage<>(map)
```
is enough.